activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -231,12 +232,19 @@ def read_multi(*names)
       # Failsafe: Raises errors.
       def delete_matched(matcher, options = nil)
         instrument :delete_matched, matcher do
-          case matcher
-          when String
-            redis.with { |c| c.eval DELETE_GLOB_LUA, [], [namespace_key(matcher, options)] }
-          else
+          unless String === matcher
             raise ArgumentError, "Only Redis glob strings are supported: #{matcher.inspect}"
           end
+          redis.with do |c|
+            start = "0"
+            pattern = namespace_key(matcher, options)
+            # Fetch keys in batches using SCAN to avoid blocking the Redis server.
+            while true
```

Comment:
In theory, yes (no way it around that I know of).

In practice, I don't think that's possible:
1. `SCAN` is fast, 1 million records in less than a second on 2011 consumer laptop hardware.
2. Generating cache keys is not fast (requires doing something that needs caching first) and at least 1,000 cache writes per second would need to be happening assuming 1 million keys and 2011-level laptop hardware (likely many more, because modern server hardware is faster).